docs/source/tutorials_source/platform/tutorial_active_learning_detectron2.py

```diff
@@ -0,0 +1,306 @@
+    im = cv2.imread(filename)
+    out = model(im)
+    # We can use `Visualizer` to draw the predictions on the image.
+    v = Visualizer(im[:, :, ::-1], MetadataCatalog.get(cfg.DATASETS.TRAIN[0]), scale=1.2)
+    out = v.draw_instance_predictions(out["instances"].to("cpu"))
+    plt.figure(figsize=(16,12))
+    plt.imshow(out.get_image()[:, :, ::-1])
+    plt.axis('off')
+    plt.tight_layout()
+
+# %%
+# The lightly framework expects a certain bounding box and prediction format. 
+# We create another helper method to convert the detectron2 output into 
+# the desired format.
+def convert_bbox_detectron2lightly(outputs):
+    # convert detectron2 predictions into lightly format
+    height, width = outputs['instances'].image_size
+    boxes = []
+
+    for (bbox_raw, score, class_idx) in zip(outputs['instances'].pred_boxes.tensor, outputs['instances'].scores, outputs['instances'].pred_classes):
+        x0, y0, x1, y1 = bbox_raw.cpu().numpy()
+        x0 /= width
+        y0 /= height
+        x1 /= width
+        y1 /= height
+      
+        boxes.append(BoundingBox(x0, y0, x1, y1))
+    output = ObjectDetectionOutput.from_scores(boxes, outputs['instances'].scores.cpu().numpy(), outputs['instances'].pred_classes.cpu().numpy().tolist())
+    return output
+
+# %%
+# Run model predictions
+# ----------------------
+#
+# We now use the created model and iterate over the `query_set` and make predictions.
+# It's important that the predictions are in the same order as the filenames
+# in the `query_set`. Otherwise, we could upload a prediction to the wrong sample!
+
+obj_detection_outputs = []
+pbar = tqdm.tqdm(al_agent.query_set)
+for fname in pbar:
+  fname_full = os.path.join(DATASET_ROOT, fname)
+  im = cv2.imread(fname_full)
+  out = predictor(im)
+  obj_detection_output = convert_bbox_detectron2lightly(out)
+  obj_detection_outputs.append(obj_detection_output)
+
+# %%
+# Now, we need to turn the predictions into scores.
+# The scorer assigns scores between 0.0 and 1.0 to 
+# each sample and for each scoring method.
+scorer = ScorerObjectDetection(obj_detection_outputs)
+scores = scorer.calculate_scores()
+# %% 
+# Let's have a look at the sample with the highest
+# prediction-margin score.
+max_score = scores['uncertainty_margin'].max()
+idx = scores['uncertainty_margin'].argmax()
+print(f'Highest uncertainty_margin score found for idx {idx}: {max_score}')
+
+# %%
+# Let's have a look at this particular image and show the model 
+# prediction for it.
+fname = os.path.join(DATASET_ROOT, al_agent.query_set[idx])
+predict_and_overlay(predictor, fname)
+
+# %%
+# Finally, we can tell our agent to select the top 100 images to annotate and
+# improve our existing model. We pick the sampling method called `CORAL` which
+# is a combination of Coreset and Active Learning. Whereas Coreset maximizes
+# the image diversity based on the embeddings, active learning aims at selecting
+# images where our models struggles the most. 
```

Comment:
One more sentence about why this is a good thing? :)

Reply:
E.g. 
"To understand why it is helpful to combine Active Learning with CORESET, consider the following case: There are two very similar images, both of which have a high uncertainty score. Having very similar samples does not help the model to generalise well, thus it would be good, if only one of these samples was chosen.
 A pure Active Learning sampler, however, would choose both of them, whereas CORAL also takes the diversity of samples into account and thus only chooses one of them.